hydroflow/examples/chat/main.rs

```diff
@@ -42,12 +42,17 @@ async fn main() {
 
     match opts.role {
         Role::Client => {
-            let client_addr = opts.client_addr.unwrap();
+            // allocate `outbound` sink and `inbound` stream
+            let client_addr = match opts.client_addr {
+                Some(addr) => addr,
+                None => ipv4_resolve(format!("localhost:{}", 0).as_str()).unwrap(),
```

Comment:
`format!("localhost:{}", 0).as_str()` -> `"localhost:0"`

Kinda surprised this mechanism works to pick a port, neat

Reply:
Actually also this can be written with an `unwrap_or_else`

`let client_addr = opts.client_addr.unwrap_or_else(|| ipv4_resolve("localhost:0").unwrap())`

Reply:
port 0 is standard for "gimme a port!"